controller/error_pages.rst

```diff
@@ -39,45 +39,51 @@ Error pages for the production environment can be customized in different ways
 depending on your needs:
 
 #. If you just want to change the contents and styles of the error pages to match
-   the rest of your application, :ref:`override the default error templates <use-default-exception-controller>`;
+   the rest of your application, :ref:`override the default error templates <use-default-error-controller>`;
 
 #. If you also want to tweak the logic used by Symfony to generate error pages,
-   :ref:`override the default exception controller <custom-exception-controller>`;
+   :ref:`override the default exception controller <custom-error-controller>`;
 
 #. If you need total control of exception handling to execute your own logic
    :ref:`use the kernel.exception event <use-kernel-exception-event>`.
 
-.. _use-default-exception-controller:
-.. _using-the-default-exceptioncontroller:
+.. _use-default-error-controller:
+.. _using-the-default-errorcontroller:
 
 Overriding the Default Error Templates
 --------------------------------------
 
-When the error page loads, an internal :class:`Symfony\\Bundle\\TwigBundle\\Controller\\ExceptionController`
+The error templates are rendered using the ``error_renderer`` service.
+You can use the built-in Twig error renderer for HTML error pages or create your own renderer
+for other formats like JSON.
```

Comment:
Aha I see. I wasn't aware of the other Error Renderers. It would seem right to document at least how to use the SerializerErrorRenderer to render a json error for example. 

> For instance, the Serializer component has a built-in FlattenException normalizer (ProblemNormalizer) and JSON/XML/CSV/YAML encoders by default, i.e. you won't have to do anything more if you want to get the error in JSON format.

So how does this work in practice. Let's say I'd want the following output:
```json
{
    "content": "Something went wrong.",
    "metadata": {
        "exception": {
            "message": "$exception->getMessage()",
            "code": "$exception->getCode()"
        }
    }
}
```

Could you maybe provide a small code example of what would be required to achieve this?

I think based on that example I would understand enough to be able to write a paragraph about the `SerializerErrorRenderer`.

Reply:
Yes, if you want a different error content/structure for your API, just create a new normalizer, see a simple example in the [upgrade notes](https://github.com/symfony/symfony/blob/4.4/UPGRADE-4.4.md#twigbundle):
```php
class MyProblemNormalizer implements NormalizerInterface
{
    public function normalize($exception, $format = null, array $context = [])
    {
        return [
            'content': 'Something went wrong.',
            'metadata': [
                'exception': [
                    'message': $exception->getMessage(),
                    'code': $exception->getStatusCode(),
                ],
            ],
        ];
    }

    public function supportsNormalization($data, $format = null)
    {
        return $data instanceof FlattenException;
    }
}
```

Reply:
Thanks! I shall write a paragraph about this perhaps tomorrow.

Reply:
Thank you!